### DISCUSSION TITLE
How to import a csv file into postgres using prisma

### DISCUSSION THREAD

**Discussion by sdq-sts** [prisma/prisma#16470] [Q&A]
Hello guys

So I'm using nestjs, docker, Prisma, and Postgres if that matters.

I have a container running Postgres and I want to seed some data from a `.csv` file but I get an error message when trying to use the seed feature, that's how it looks:

```js
// seed.ts
const accountSeedFile = join(process.cwd(), './prisma/seeds/accounts.csv');

const accountSeedPromise = prisma.$executeRaw`
  COPY "accounts"(id, name, bank)
  FROM ${accountSeedFile}
  DELIMITER ','
  CSV HEADER;
`;

const [accountsResult] = await Promise.all([accountSeedPromise]);
```

When I try to use `prisma db seed` I get the following:

```sh
code: 'P2010',
clientVersion: '4.6.1',
meta: {
  code: '42601',
  message: 'db error: ERROR: syntax error at or near "$1"'
}
```

Even if I can execute this command I think I'll not have the expected results because I don't think I'll have the right to open the file from Prisma.

Am I missing something here?
Is that the right way to seed a database from a `.csv` file?

**Comment by pantharshit00** [comment]
`$executeRaw` will turn the variable you have given into a prepared statement variable so it won't work here. You can try the following:
```ts
// seed.ts
const accountSeedFile = join(process.cwd(), './prisma/seeds/accounts.csv');

const accountSeed = await prisma.$executeRawUnsafe(`
  COPY "accounts"(id, name, bank)
  FROM ${accountSeedFile}
  DELIMITER ','
  CSV HEADER;
`);

```

Please do note that this is prone to SQL injection if `accountSeedFile` is not controlled by you.

**Reply by vladi-stevanovic** [reply]
👋 Hi @sdq-sts , did you perhaps have a chance to check Harshit's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Reply by pedropcruz** [reply]
I used your suggestion @pantharshit00 and I got a similar error that the previous one.

``` {
  code: 'P2010',
  clientVersion: '4.7.1',
  meta: {
    code: '42601',
    message: 'db error: ERROR: syntax error at or near "C"'
  },
  batchRequestIdx: undefined
}
```

**Reply by pantharshit00** [reply]
Hey @pedropcruz,

Will it be possible if you can share the csv and the schema file here? That way I can try it out myself.

**Reply by djarran** [reply]
Tried this answer. And get the following error

```
must be superuser or have privileges of the pg_read_server_files role to COPY from a file
HINT: Anyone can COPY to stdout or from stdin. psql's \copy command also works for anyone.
```

I have also tried using this answer [here](https://stackoverflow.com/a/30951435) and it does not work with the following error: ERROR: syntax error at or near "\

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
